ci-automation/vendor-testing/qemu_update.sh

```diff
@@ -67,14 +72,40 @@ if [ "${CIA_ARCH}" = "arm64" ]; then
     fi
 fi
 
-kola run \
-    --board="${CIA_ARCH}-usr" \
-    --parallel="${QEMU_PARALLEL}" \
-    --platform=qemu \
-    --qemu-bios="${bios}" \
-    --qemu-image=tmp/flatcar_production_image_previous.bin \
-    --tapfile="${CIA_TAPFILE}" \
-    --torcx-manifest="${CIA_TORCX_MANIFEST}" \
-    --update-payload=tmp/flatcar_test_update.gz \
-    --qemu-skip-mangle \
-    cl.update.payload
+query_kola_tests() {
+    shift; # ignore the instance type
+    local arg
+    arg="${*}"
+    if [ "${arg}" != "" ]; then
+      arg="cl.update.payload"
+    fi
+    kola list --platform=qemu --filter "${arg}"
+}
+
+run_kola_tests() {
+    local instance_type="${1}"; shift;
+    local instance_tapfile="${1}"; shift
+    local image
+    if [ "${instance_type}" = "previous" ]; then
+        image="tmp/flatcar_production_image_previous.bin"
+    elif [ "${instance_type}" = "first_dual" ]; then
+        image="tmp/flatcar_production_image_first_dual.bin"
+    else
+        echo "Wrong instance type ${instance_type}" >&2
+        exit 1
+    fi
+
+    kola run \
+      --board="${CIA_ARCH}-usr" \
+      --parallel="${QEMU_PARALLEL}" \
+      --platform=qemu \
+      --qemu-bios="${bios}" \
+      --qemu-image="${image}" \
+      --tapfile="${instance_tapfile}" \
+      --torcx-manifest="${CIA_TORCX_MANIFEST}" \
+      --update-payload=tmp/flatcar_test_update.gz \
+      --qemu-skip-mangle \
+      cl.update.payload
```

Comment:
Makes no difference here, but…

```suggestion
      "${@}"
```

Reply:
For some reason I thought that "${@}" will be "filtered" by `query_kola_tests`, but that happens only for extra instances. Nevermind.